client/app/dashboard/model/list/ModelList.js

```diff
@@ -713,7 +713,10 @@ angular.module('dashboard.Dashboard.Model.List', [
   
   $scope.$watch('pagingOptions', function (newVal, oldVal) {
     if (newVal.currentPage != oldVal.currentPage || newVal.pageSize != oldVal.pageSize) {
+      if (!newVal.currentPage) newVal.currentPage = 1;
       $scope.pagingOptions.pageSize = $scope.pagingOptions.pageSize.toString();
+      var maxPageNumber = Math.ceil($scope.totalServerItems / newVal.pageSize);
+      if (newVal.currentPage > maxPageNumber) newVal.currentPage = maxPageNumber;
```

Comment:
Should we check to see if newVal.currentPage != oldVal.currentPage again so we don't call $scope.loadItems() unnecessarily?

Or refactor so we only do that comparison once?

Reply:
I think we don't need to because we already check that on line 716.

Reply:
It is difficult to type in a page number that doesn't start with 1 (like page 20).

Inside of live updating, would it be better to just wait for an <enter>?

Reply:
I will use debounce then instead.